clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java

```diff
@@ -1120,13 +1127,17 @@ protected static boolean shouldBuildZstdCompressionDictionary(
       return false;
     }
 
+    if (pushJobSetting.isIncrementalPush) {
+      LOGGER.info("No compression dictionary will be generated as the push type is incremental push");
+      return false;
+    }
+
+    if (!inputFileHasRecords) {
```

Comment:
This could be a duplicate log as we check and log it later in the function as well.